tests/test_inventory_events_consumer.py

```diff
@@ -103,11 +106,55 @@ def test_host_update_event_no_cp(inventory_event_consumer, inventory_event_messa
         assert updated_system.display_name != updated_display_name
 
 
-def test_host_delete_event(inventory_event_consumer, db_setup):
-    msg = {"type": "delete", "insights_id": "677fb960-e164-48a4-929f-59e2d917b444",
+def test_host_delete_event(inventory_event_consumer, inventory_event_message, db_setup, redis_setup, mocker):
+    mocker.patch.object(
+        inventory_event_consumer,
+        'process_system_details',
+        side_effect=inventory_event_consumer.process_system_details,
+        autospec=True
+    )
+    inventory_event_message['type'] = 'created'
+    inventory_event_consumer.host_create_update_events(inventory_event_message)
+
+    msg = {"type": "delete",
+           "insights_id": "677fb960-e164-48a4-929f-59e2d917b444",
            "id": "ee0b9978-fe1b-4191-8408-cbadbd47f7a2",
-           "account": '0000001',
-           'org_id': '000001'}
+           "account": '0000001', 'org_id': '000001'}
     inventory_event_consumer.host_delete_event(msg)
+    with app.app_context():
+        cached_deleted_sys = cache.get(
+            f"{msg['org_id']}{CACHE_KEYWORD_FOR_DELETED_SYSTEM}{msg['id']}"
+        )
+        assert cached_deleted_sys == 1
+
     host = db_get_host(msg['id'])
     assert host is None
+
+
+def test_recreate_remove_cached_key(inventory_event_consumer, inventory_event_message, db_setup, redis_setup, mocker):
```

Comment:
Do we need `redis_setup` in the other tests?

Reply:
Hello @saltgen,
  I verified all tests, removed redis_client fixture where it wasn't needed. 
  On Engine processor, found only one test which directly calls `handle_msg` method that uses `cache` .
  On Inventory processor side, also removed fixture. 
  
  Thank you for your suggestion.
  
  ```
  def _connect(self):
        "Create a TCP socket connection"
        # we want to mimic what socket.create_connection does to support
        # ipv4/ipv6, but we want to set options prior to calling
        # socket.connect()
        err = None
        for res in socket.getaddrinfo(
            self.host, self.port, self.socket_type, socket.SOCK_STREAM
        ):
            family, socktype, proto, canonname, socket_address = res
            sock = None
            try:
                sock = socket.socket(family, socktype, proto)
                # TCP_NODELAY
                sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
    
                # TCP_KEEPALIVE
                if self.socket_keepalive:
                    sock.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
                    for k, v in self.socket_keepalive_options.items():
                        sock.setsockopt(socket.IPPROTO_TCP, k, v)
    
                # set the socket_connect_timeout before we connect
                sock.settimeout(self.socket_connect_timeout)
    
                # connect
>               sock.connect(socket_address)
E               ConnectionRefusedError: [Errno 111] Connection refused

../../../.cache/pypoetry/virtualenvs/ros-backend-ddIXnhYv-py3.11/lib/python3.11/site-packages/redis/connection.py:627: ConnectionRefusedError

During handling of the above exception, another exception occurred:

engine_result_message = {'input': {'host': {'account': '0000001', 'ansible_host': None, 'bios_uuid': 'ec2c79ab-02ff-d09b-31a9-583a902dd74b', '...ening_tls_key_perms.report', 'details': {'pass_key': 'OK', 'type': 'pass'}, 'key': 'OK', 'links': {}, ...}, ...], ...}}
engine_consumer = <ros.processor.insights_engine_consumer.InsightsEngineConsumer object at 0x7f7abadcc490>
mocker = <pytest_mock.plugin.MockerFixture object at 0x7f7ab990f850>
performance_record = {'disk.dev.total': {'nvme0n1': {'units': 'count / sec', 'val': 7.22}}, 'hinv.ncpu': 8.0, 'instance_type': 't2.micro', 'kernel.all.cpu.idle': 7.55, ...}

    def test_handle_msg(engine_result_message, engine_consumer, mocker, performance_record):
        engine_result_message = engine_result_message("insights-engine-result-idle.json")
        mocker.patch(
            'ros.processor.insights_engine_consumer.get_performance_profile',
            return_value=performance_record,
            autospec=True
        )
        mocker.patch.object(engine_consumer, 'process_report', return_value=True, autospec=True)
>       engine_consumer.handle_msg(engine_result_message)

tests/test_insights_engine_result_consumer.py:62: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ros/processor/insights_engine_consumer.py:102: in handle_msg
    if cache.get(cache_key):
../../../.cache/pypoetry/virtualenvs/ros-backend-ddIXnhYv-py3.11/lib/python3.11/site-packages/flask_caching/__init__.py:195: in get
    return self.cache.get(*args, **kwargs)
```